app/blocked/api/route.ts

```diff
@@ -1,20 +1,14 @@
 import { NextRequest, NextResponse } from "next/server";
+import { checkOFAC } from "./util";
 
 export const dynamic = "force-dynamic"; // Always run dynamically
 
 export async function POST(request: NextRequest) {
-  // Input data
-  /*
-    {
-      "sourceAddress": "0x00000000...."
-      "beneficiary": "0x00000000...."
-    }
-
-  */
   const { sourceAddress, beneficiaryAddress } = await request.json();
-  // TODO: Lookup address
+  const sourceBanned = await checkOFAC(sourceAddress);
+  const beneficiaryBanned = await checkOFAC(beneficiaryAddress);
   return NextResponse.json({
-    sourceBanned: true,
-    beneficiaryBanned: true,
+    sourceBanned: sourceBanned,
+    beneficiaryBanned: beneficiaryBanned,
```

Comment:
i think need to still use these vars in the middleware to return blocked page if they're true

Reply:
We cannot do this in middleware as the addresses are not in any HTTP requests (client side only). We need to modify the `components/transfer.tsx` to call this endpoint before allowing a transaction to be submitted onchain from the client side js.

Here is where to add the code in the `doSubmit` method. This also shows how to trigger validation error with ` form.setError`.
https://github.com/Snowfork/snowbridge-app/blob/5f41ec3978a07e25d675656ad5a5555784ceea8b/components/transfer.tsx#L508-L512

The `data: FormData` has the addresses. You can use `data.sourceAccount` and `data.beneficiary` and validate those.